routes/api/users.js

```diff
@@ -180,6 +183,55 @@ router.patch("/avatars", [authToken, userLoggedIn, upload.single("avatar")], asy
     }
 });
 
+router.get("/verify/:verificationToken", async (req, res, next) => {
+  try {
+    const { verificationToken } = req.params;
+    const user = await Users.findOneAndUpdate(
+      { verificationToken },
+      { verify: true, verificationToken: null },
+      { new: true }
+    );
+    console.log(user);
+
+    if (user) {
+      return res.status(200).json({ message: "Verification successful" });
+    } else {
+      return res.status(404).json({ message: "User not found" });
+      next();
+    }
+  } catch (error) {
+    console.error("Something went wrong: ", error);
+    next(error);
+  }
+});
+
+router.post("/verify", async (req, res, next) => {
+  try {
+    const { email } = req.body;
+    const user = await Users.findOne({ email });
+
+    if (!email) {
+      return res
+        .status(400)
+        .json({ message: `Missing required field email` });
+    }
+
+    if (user.verify) {
+      return res
+        .status(400)
+        .json({ message: `Verification has already been passed` });
+    }
+
+    mailer.sendVerificationEmail(user.email, user.verificationToken);
```

Comment:
Tutaj dobrze byłoby wiedzieć czy email naprawdę został wysłany ponieważ może być sytuacja że będziesz miał błąd a ty wysyłasz informacje do użytkownika (poniżej) że wiadomość została wysłana. Dlatego Kiedy przejdziesz na składnie async/await to będziesz mógł ewentualnie złapać ten błąd.